ppa/archive/forms.py

```diff
@@ -25,9 +25,31 @@ def valid_value(self, value):
 
 class SearchForm(forms.Form):
     '''Simple search form for digitized works.'''
+
+    SORT_CHOICES = [
+        ('relevance', 'Relevance'),
+        ('pub_date_asc', 'Year Newest-Oldest'),
+        ('pub_date_desc', 'Year Oldest-Newest'),
+        ('title_asc', 'Title A-Z'),
+        ('title_desc', 'Title Z-A'),
+        ('author_asc', 'Author A-Z'),
+        ('author_desc', 'Author Z-A'),
+    ]
+
+    def __init__(self, *args, **kwargs):
+        '''
+        Set choices dynamically based on form kwargs and presence of keywords.
+        '''
+        super(SearchForm, self).__init__(*args, **kwargs)
+        if not args or 'query' not in args[0] or not args[0]['query']:
```

Comment:
I like this solution, not what I had thought of.  Does it impact the design at all?  I had imagined relevance being listed but greyed out, but now that I think about it I don't know how that would work.

Reply:
@thatbudakguy @xinyil A quick question on this for when I pick this back up and not wholly clear from the design spec (as far as I can see): when relevance isn't a valid option (i.e. no keyword search made), should it be in the form choices but disabled in some way or just not there at all (how I have it currently written)?

Reply:
I kind of like it being disabled, personally. moves less things around and easier to indicate in the UI.

Reply:
Kind of a bummer that we have to do that, but I think it'd work. As long as it supports disabling I don't think the widget type matters much because I have to iterate through this thing to make it display correctly (it's not going to render as radio buttons or a select anyway).